Xamarin.Forms.Core/Items/ItemsView.cs

```diff
@@ -97,4 +98,29 @@ protected virtual void OnScrollToRequested(ScrollToRequestEventArgs e)
 			ScrollToRequested?.Invoke(this, e);
 		}
 	}
+
+	public enum SelectionMode 
+	{
+		None,
+		Single,
+		Multiple
+	}
+
+	public class SelectionChangedEventArgs : EventArgs
+	{
+		public IReadOnlyList<object> PreviousSelection { get; }
+		public IReadOnlyList<object> CurrentSelection { get; }
+
+		public SelectionChangedEventArgs(object previousSelection, object currentSelection)
```

Comment:
as this one is a convenience helper, I'd make it internal

Reply:
> as this one is a convenience helper, I'd make it internal

Done.

>  move SelectionMode enum and SelectionChangedEventArgs class

Moved them to their own files.